src/bpffeature.cpp

```diff
@@ -232,6 +233,29 @@ int BPFfeature::instruction_limit(void)
   return *insns_limit_;
 }
 
+bool BPFfeature::has_d_path(void)
+{
+  if (has_d_path_.has_value())
+    return *has_d_path_;
+
+  struct bpf_insn insns[] = {
```

Comment:
Why does this need a custom program? doesn't the helper detection work here?

Reply:
aah, I try to load the working program and check that try_load succeeds,
but I did not noticed that the detection stuff checks for 'invalid' string,
so the load can fail for other reasons but still detect the helper.. that might
actually work.. I'll try it ;-)

thanks

Reply:
hm, the tracing program fails earlier with:
  Tracing programs must provide btf_id

so we need to have special function for that